app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,19 @@
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by email: params[:session][:email].downcase
+    if user&.authenticate params[:session][:password]
+      log_in user
+      redirect_to user
+    else
+      flash.now[:danger] = t ".invalid"
+      render :new
+    end
+  end
+
+  def destroy
+    log_out
```

Comment:
nên check xem đã login chưa rồi xử lý
```suggestion
    log_out if logged_in?
```